katdal/sensordata.py

```diff
@@ -462,6 +462,73 @@ def _safe_linear_interp(xi, yi, x):
     return (1.0 - end_weight) * yi[start] + end_weight * yi[end]
 
 
+_KATCP_DECODER = {'integer': int, 'float': float, 'boolean': lambda x: x == '1',
+                  'lru': str, 'timestamp': float, 'discrete': str,
+                  'address': str, 'string': str}
+
+
+def get_sensor_from_katstore(store, name, start_time, end_time):
+    """Get raw sensor data from katstore (CAM's central sensor database).
+
+    Parameters
+    ----------
+    store : string
+        Hostname / endpoint of katstore webserver speaking katstore REST API
+    name : string
+        Sensor name (the escaped version with underscores)
+    start_time, end_time : float
+        Time range for sensor records as UTC seconds since Unix epoch
+
+    Returns
+    -------
+    data : :class:`RecordSensorData` object
+        Retrieved sensor data with 'timestamp', 'value' and 'status' fields
+
+    Raises
+    ------
+    ConnectionError
+        If this cannot connect to the katstore server
+    ValueError
+        If connection succeeded but interaction with REST API failed
+    KeyError
+        If the sensor was not found in the store or it has no data
+    """
+    # First check existence of sensor and get its KATCP type
+    url = "http://{}/katstore/sensors".format(store)
+    try:
+        result = requests.get(url, params={'sensors': name})
```

Comment:
Is that a Python regular expression?

Reply:
No, it gets handled by Postgres.  Part of a SQL statement like ` "WHERE name ~* %s "`

Reply:
So I assume the anchors are `^` and `$`?

Reply:
I assume so too, but best to test it.

Reply:
You should also escape the contents as well. I guess the fun part will be telling whether `re.escape` is going to escape the right things for Postgres.

Reply:
I'm just doing what `katportalclient` would have done here.

Reply:
The anchoring and escaping doesn't seem to have been implemented.

Reply:
I didn't implement it because I took katportalclient as the gold standard and it didn't do it. But if you insist... :-) I've now added a sanitisation step before the HTTP request which should at least avoid the worst offenses.